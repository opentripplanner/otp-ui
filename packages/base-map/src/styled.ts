import { Popup as MapGlPopup } from "react-map-gl";
import styled from "styled-components";

/**
 * Adds a box shadow and tweaks border radius to make popups easier to read.
 */
export const Popup = styled(MapGlPopup)`
  & > .maplibregl-popup-content,
  & > .mapboxgl-popup-content {
    border-radius: 10px;
    box-shadow: 0 3px 14px 4px rgb(0 0 0 / 20%);
  }
`;

export const MapOverlayPopup = styled.div`
  font-size: 12px;
  line-height: 1.5;
  min-width: 250px;
`;

export const PopupRow = styled.p`
  margin-top: 6px;
`;

export const PopupTitle = styled.header`
  font-size: 18px;
  font-weight: 500;
  margin-bottom: 6px;
`;

type LeafletStyleMarkerProps = {
  color?: string;
  size?: number;
  stroke?: number;
  strokeColor?: string;
};
/**
 * @deprecated this marker was created to make the transition from Leaflet more manageable,
 * but in most cases this marker should not be used -- use a MapLibreGL Circle instead
 *
 * https://maplibre.org/maplibre-gl-js-docs/style-spec/layers/#circle
 */
export const LeafletStyleMarker = styled.div<LeafletStyleMarkerProps>`
  background: ${props => props?.color || "#0000ff"}50;
  border-radius: ${props => props?.size || 10}px;
  border: ${props => props?.stroke || 2}px solid
    ${props => props?.strokeColor || `${props?.color}f0` || "#0000fffo"};
  content: "";
  cursor: pointer;
  display: block;
  height: ${props => props?.size || 10}px;
  width: ${props => props?.size || 10}px;
`;

export const LayerSelector = styled.aside`
  display: flex;
  justify-content: end;
  margin: 10px;
  position: relative;
  right: 0;
  top: 0;

  /* 
  There are some situations when using MapLibreGL's \`reuseMap\` where 
  a map re-render can cause the layer selector to be rendered multiple times. 
  This is a bit of a hack, but the most all-encompassing way to ensure that this is
  not a usability issue.

  This hack *only* works if the LayerSelector is *not* a \`div\`. I'm not sure why...
  */
  &:not(:last-of-type) {
    display: none;
  }

  .layers-list {
    background: rgba(255, 255, 255, 0.95);
    list-style-type: none;
    padding: 1em;
    position: absolute;
    right: 0;
    z-index: 1000;

    label {
      display: block;
      height: 0;
      overflow: hidden;
      width: 0;

      input {
        margin-right: 1ch;
      }
    }

    &::after {
      display: block;
      content: "üåê";
      cursor: pointer;
    }

    &:hover,
    &.fake-mobile-hover {
      box-shadow: 0px -1px 15px -3px rgba(0, 0, 0, 0.1);
      label {
        height: unset;
        overflow: unset;
        width: unset;
      }
      li:first-child {
        /* Prevent most accidental toggling of the first selectable layer on touch screens.
           Toggling is still possible if you touch the left-most area of the globe "button". */
        margin-right: 2em;
      }
      &::after {
        display: none;
      }
    }
  }
`;

/**
 * Map container for use with storybook across various packages in this repo.
 */
export const StoryMapContainer = styled.div`
  height: 90vh;
`;
