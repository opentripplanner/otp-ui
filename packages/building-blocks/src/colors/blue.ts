const blue = {
  50: "#E5ECF3",
  100: "#CCDAE7",
  200: "#B2C7DB",
  300: "#99B5CF",
  400: "#7FA2C2",
  500: "#6690B7",
  600: "#4D7EAB",
  // 700 and above text is WCAG AA accessible on white backgrounds.
  700: "#336B9E",
  // 800 and above text is WCAG AAA accessible on white backgrounds.
  800: "#1A5992",
  900: "#004686"
};

export default blue;
