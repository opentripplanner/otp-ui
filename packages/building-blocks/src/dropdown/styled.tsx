import styled from "styled-components";
import grey from "../colors/grey";

export const DropdownButton = styled.button`
  background: #fff;
  border: 1px solid black;
  border-radius: 5px;
  color: inherit;
  padding: 5px 7px;
  transition: all 0.1s ease-in-out;

  span.caret {
    border-left: 4px solid transparent;
    border-right: 4px solid transparent;
    border-top: 4px solid;
    color: inherit;
    display: inline-block;
    height: 0;
    margin-left: 5px;
    vertical-align: middle;
    width: 0;
  }

  &:hover,
  &[aria-expanded="true"] {
    background: ${grey[50]};
    color: black;
    cursor: pointer;
  }
`;

export const DropdownMenu = styled.div<{ alignLeft?: boolean }>`
  background-clip: padding-box;
  background-color: #fff;
  border-radius: 4px;
  border: 1px solid rgba(0, 0, 0, 0.15);
  box-shadow: 0 6px 12px rgba(0, 0, 0, 0.175);
  color: #333;
  list-style: none;
  margin: 2px 0 0;
  min-width: 160px;
  padding: 5px 0;
  position: absolute;
  ${props => (props.alignLeft ? "left: 0;" : "right: 0;")}
  top: 100%;
  width: 100%;
  z-index: 1000;

  hr {
    margin: 0;
    padding: 0;
  }
  a,
  button {
    background: transparent;
    border: none;
    cursor: pointer;
    padding: 5px 15px;
    text-align: start;
    width: 100%;

    &:hover {
      background: ${grey[50]};
    }
  }
`;

export const DropdownWrapper = styled.span<{ pullRight?: boolean }>`
  float: ${props => (props.pullRight ? "right" : "left")};
  position: relative;
`;
