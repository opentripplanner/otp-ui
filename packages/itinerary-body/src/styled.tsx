// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore FIXME: Create TypeScript types for the icons package.
import { Map } from "@opentripplanner/icons";
import React, { ReactElement } from "react";
import styled from "styled-components";
import { CaretDown } from "@styled-icons/fa-solid/CaretDown";
import { CaretUp } from "@styled-icons/fa-solid/CaretUp";
import { ExclamationTriangle } from "@styled-icons/fa-solid/ExclamationTriangle";

import { toModeBorder, toModeColor, toSafeRouteColor } from "./util";

interface LightBorderDivProps {
  hideBorder: string;
  theme?: {
    borderColor?: string;
  };
}

/**
 * This component is needed to give the offset border look to stacked place rows
 * Since the value we have access to is "interlineWithPreviousLeg" then we
 * have to show/hide the top border of the div and apply a small offset
 */
export const LightBorderDiv = styled.div<LightBorderDivProps>`
  border-top-style: solid;
  border-top-width: ${props => (props.hideBorder === "true" ? "0" : "2px")};
  border-top-color: ${props => props.theme.borderColor};
  padding-top: ${props => (props.hideBorder === "true" ? "0" : "10px")};
  padding-bottom: ${props => (props.hideBorder === "true" ? "10px" : "0")};
  transform: ${props =>
    props.hideBorder === "true" ? "" : "translateY(-12px)"};
`;

export const TransparentButton = styled.button`
  background: transparent;
  border: 0;
  color: inherit;
  cursor: pointer;
  font-size: inherit;
  text-decoration: none;
`;

export const AnchorButton = styled.a`
  background-color: #fff;
  background-image: none;
  border-radius: 4px;
  border: 1px solid #ccc;
  color: #333;
  cursor: pointer;
  display: inline-block;
  font-size: 14px;
  font-weight: 400;
  padding: 4px 6px;
  text-align: center;
  text-decoration: none;
  touch-action: manipulation;
  user-select: none;
  white-space: nowrap;

  &:hover {
    color: #333;
    background-color: #e6e6e6;
    border-color: #adadad;
  }

  &:active {
    color: #333;
    background-color: #e6e6e6;
    background-image: none;
    border-color: #adadad;
    outline: 0;
    box-shadow: inset 0 3px 5px rgba(0, 0, 0, 0.125);
  }

  &:focus {
    color: #333;
    background-color: #e6e6e6;
    border-color: #8c8c8c;
  }

  &:active:hover {
    color: #333;
    background-color: #d4d4d4;
    border-color: #8c8c8c;
  }
`;

export const LinkButton = styled.a`
  color: #008;
  cursor: pointer;
  margin-left: 5px;

  &:hover {
    text-decoration: underline;
  }
`;

export const ViewerButton = styled(LinkButton)`
  padding-left: 0px;

  &:before {
    content: "|";
    color: black;
    margin-right: 5px;
  }
`;

// ////////////////////////////////////////////////
// /////////////// App components /////////////////
// ////////////////////////////////////////////////

interface ModeRouteProps {
  mode: string;
  routeColor: string;
}

// TODO: Can we turn this into a more abstract element to inherit from for other badges?
export const AccessBadge = styled.div<ModeRouteProps>`
  color: black;
  background-color: ${props => toModeColor(props.mode, props.routeColor)};
  border: 2px solid #bbb;
  text-align: center;
  width: 25px;
  height: 25px;
  font-size: 1.2em;
  border-radius: 50%;
  display: flex;
  align-items: center;
  justify-content: center;
  padding-left: 1px;
  /* Add in border for dark mode */
`;

export const CallAheadWarning = styled.div`
  color: #b22727;
  margin-top: 5px;
`;

export const BookLaterContainer = styled.div`
  bottom: 0;
  left: 110px;
  position: absolute;
  right: 0;
  top: 0;
`;

export const BookLaterInnerContainer = styled.div`
  background-color: #fcf9d3;
  display: table;
  height: 100%;
  width: 100%;
`;

export const BookLaterPointer = styled.div`
  border-bottom: 16px solid transparent;
  border-right: 16px solid #fcf9d3;
  border-top: 16px solid transparent;
  height: 0;
  left: 94px;
  position: absolute;
  top: 0;
  width: 0;
`;

export const BookLaterText = styled.div`
  color: #444;
  display: table-cell;
  font-style: italic;
  line-height: 0.95;
  padding: 0px 2px;
  vertical-align: middle;
`;

export const BookTNCRideButton = styled(AnchorButton)``;

export const BookTNCRideButtonContainer = styled.div`
  height: 32px;
  margin-bottom: 10px;
  margin-top: 10px;
  position: relative;
`;

export const TNCTravelTime = styled.div`
  /* no styling */
`;

export const TNCCost = styled.div`
  /* no styling */
`;

interface CaretToggleProps {
  className?: string;
  expanded: boolean;
}

export const CaretToggleBase = ({
  className,
  expanded
}: CaretToggleProps): ReactElement => (
  <span className={className}>
    {expanded ? <CaretUp size={15} /> : <CaretDown size={15} />}
  </span>
);

export const CaretToggle = styled(CaretToggleBase)`
  &::before {
    content: "";
    margin: 0 0.125em;
  }
`;

export const Destination = styled.div`
  text-align: center;
`;

export const InnerLine = styled.div<ModeRouteProps>`
  /* the actual line element */
  border-left: ${props => toModeBorder(props.mode, props.routeColor)};
  height: 100%;
  width: 0;
  position: absolute;
  left: 50%;
  transform: translateX(-50%);
`;

export const InterlineDot = styled.div`
  color: #fff;
  flex: 0 0 15px;
  margin-left: -33px;
  margin-right: 18px;
  position: relative;
  z-index: 30;
`;

export const InterlineName = styled.div`
  /* special messaging, not sure yet */
`;

export const IntermediateStops = styled.ol`
  display: block;
  font-size: 13px;
  list-style: none;
  padding: 0;
`;

export const ItineraryBody = styled.ol`
  list-style: none;
  padding: 0;
`;

export const LegBody = styled.div`
  color: #676767;
  font-size: 13px;
  padding-bottom: 12px;
`;

export const LegClickable = styled.div``;

/**
 * Transparent button, clickable by all, with an invisible text about zooming to a leg on the map.
 * The button sits on top of LegDescription, so that the button's text visually appears to be
 * that of LegDescription.
 */
export const LegClickableButton = styled(TransparentButton)`
  bottom: 0;
  cursor: pointer;
  left: 0;
  position: absolute;
  right: 0;
  top: 0;
  /* This is required for the entire leg to be clickable in Firefox. */
  width: 100%;
  /* Place the button just above the elevation chart, 
     so that its outline doesn't appear clipped in Chromium. */
  z-index: 1;
`;

// Use <span> for correct semantics as it is the contents of a button or a link.
export const LegDescription = styled.span`
  align-items: center;
  display: inline-flex;
  line-height: 16px;
  min-height: 31px;
  position: relative;
`;

// additional description added to ClickableLeg for screenreaders
export const InvisibleAdditionalDetails = styled.span`
  display: inline-block;
  grid-row-start: 2;
  grid-column-start: 1;
  height: 0;
  overflow: hidden;
  width: 0;
`;

export const LegDescriptionHeadsignPrefix = styled.span`
  font-weight: 200;
`;

/**
 * Lets others apply styles to the mode text in
 * "Bicycle 0.5 miles to City Hall"
 */
export const LegDescriptionMode = styled.span`
  font-weight: inherit;
`;

/**
 * Lets others apply styles to the place text in
 * "Bicycle 0.5 miles to City Hall"
 */
export const LegDescriptionPlace = styled.span`
  font-weight: inherit;
`;

export const LegDescriptionRouteLongName = styled.span`
  font-size: 13px;
  font-weight: 500;
`;

export const LegDescriptionRouteShortName = styled.span`
  font-weight: 800;
  margin-right: 6px;
`;

export const LegDescriptionForTransit = styled(LegDescription)`
  color: #807373;
  margin-top: 5px;
`;

export const LegIconContainer = styled.span`
  img,
  svg {
    margin-right: 6px;
    height: 24px;
    width: 24px;
    vertical-align: bottom;
  }
`;

export const LegIconAndRouteShortName = styled.span`
  flex-shrink: 0;
`;

export const LegLine = styled.div`
  position: relative;
  left: 50%;
  transform: translateX(-50%);
  height: 100%;
`;

export const LineBadgeContainer = styled.div`
  width: 30px;
  height: 30px;
  border-radius: 50%;
  position: absolute;
  left: 50%;
  top: 0;
  transform: translate(-51%, -10%);
`;

export const LineColumn = styled.div`
  /* flexbox column */
  grid-column-start: 2;
  grid-row: span 2;
  padding-right: 5px;
`;

export const LegDetails = styled.span`
  display: grid;
  grid-template-columns: 100px auto;
`;

export const PlaceRowWrapper = styled.li`
  /* needs to be a flexbox row */
  max-width: 500px;
  display: grid;
  grid-template-columns: 65px 30px auto;
`;

interface PreviewContainerProps {
  active: boolean;
}

export const PreviewContainer = styled.div<PreviewContainerProps>`
  background-color: ${props => props.active && "#eee"};
  border-color: ${props => (props.active ? "#d1d5da" : "#fff")};
  border-radius: 5px;
  border-style: solid;
  border-width: 1px;
  display: inline-block;
  font-style: normal;
  grid-column: 2;
  grid-row: 1;
  margin: 0 4px;
  position: relative;
  text-align: center;
  text-decoration: none;
  vertical-align: middle;
  width: 75%;

  &:hover {
    border-color: #d1d5da;
    background-color: #f6f8fa;
  }
`;

export const TimeColumn = styled.div`
  grid-column-start: 1;
  grid-row: 1 / span 2;
  padding-right: 5px;
  font-size: 0.9em;
`;

export const MapButton = styled(LinkButton)`
  padding: 3px 10px 3px 10px;
  border: 0;
  margin-top: -15px;
  width: 35px;
  height: 35px;
  &:hover {
    cursor: pointer;
  }
`;

export const MapButtonColumn = styled(LightBorderDiv)`
  flex: 0 0 25px;
  grid-column: -1;
`;

export const MapIcon = styled(Map).attrs(props => ({
  fill: props.theme.secondaryColor,
  width: 15,
  height: 15,
  role: "img"
}))``;

export const PlaceDetails = styled.div`
  grid-row-start: 2;
  grid-column-start: 3;
`;

export const PlaceHeader = styled.div`
  display: flex;
  font-size: 1.2em;
  grid-row-start: 1;
  grid-column-start: 3;
`;

export const PlaceName = styled.span`
  /* text styling */
  font-size: inherit;
  font-weight: bold;
  height: 1.2em;
  margin: 0;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  flex: 1 1 auto;
  padding: 3px 0 10px 0;
`;

export const PlaceSubheader = styled.div`
  color: #807373;
  font-size: 13px;
  font-weight: 300;
  padding-top: 1px;

  /* Reduce vertical space and fix horizontal alignment of stop id and stop viewer link for transit stops. */
  /* Also, increase vertical space after so that transit/access instructions
     aren't too squeezed with the stop id and Stop Viewer link. */
  margin-bottom: 10px;
  margin-top: -14px;
`;

export const PreviewDiagram = styled(TransparentButton)`
  padding: 2px;
  width: 100%;
`;

export const PreviewDiagramElevationChange = styled.span`
  font-size: xx-small;
  &::before {
    content: "";
    margin: 0 0.125em;
  }
`;

export const PreviewDiagramElevationGain = styled(
  PreviewDiagramElevationChange
)`
  color: #e60000;
`;

export const PreviewDiagramElevationLoss = styled(
  PreviewDiagramElevationChange
)`
  color: green;
`;

export const PreviewDiagramTitle = styled.div`
  font-size: small;
`;

interface RouteBadgeProps {
  routeColor: string;
  theme?: {
    badgeBorderColor?: string;
    mainColor?: string;
  };
}

export const RouteBadge = styled.div<RouteBadgeProps>`
  text-align: center;
  min-width: 30px;
  min-height: 30px;
  font-size: 1.2em;
  background-color: ${props =>
    toSafeRouteColor(props.routeColor) || props.theme.mainColor};
  color: white;
  border-radius: 50%;
  display: flex;
  align-items: center;
  justify-content: center;
  padding-left: 1px;
  /* Add in border for dark mode */
  border: 1px solid ${props => props.theme.badgeBorderColor};
  user-select: none;
  cursor: default;
`;

export const SROnly = styled.span`
  position: absolute;
  width: 1px;
  height: 1px;
  padding: 0;
  margin: -1px;
  overflow: hidden;
  clip: rect(0, 0, 0, 0);
  border: 0;
`;

export const SRHidden = styled.span.attrs({ "aria-hidden": true })``;

export const Steps = styled.ol`
  display: block;
  list-style: none;
  padding: 0;
`;

export const StepDescriptionContainer = styled.div`
  margin-left: 24px;
  line-height: 1.25em;
  padding-top: 1px;

  & > span {
    margin-right: 1ch;
  }
`;

export const StepsHeaderAndMapLink = styled.span`
  display: inline-flex;
  align-self: center;
  margin-top: 10px;

  /* This is needed to avoid vertical jumps on Safari and Firefox */
  a {
    display: flex;
    align-items: center;
    justify-content: center;
  }
`;

export const StepsHeader = styled(TransparentButton)`
  color: #676767;
  font-size: 13px;
  font-style: normal;
  padding: 0;
`;

export const StepIconContainer = styled.div`
  fill: #676767;
  float: left;
  height: 16px;
  width: 16px;
`;

export const StepRow = styled.li`
  font-size: 13px;
  margin-top: 8px;
  color: #676767;
  font-style: normal;
`;

export const StepStreetName = styled.span`
  font-weight: 500;
`;

export const StepLength = styled.span`
  font-weight: 200;
  /* This is the lowest opacity which still meets AA text color contrast standards */
  opacity: 0.8975;
  padding-left: 1ch;
`;

export const StopIdSpan = styled.span`
  font-weight: 200;
  font-size: 0.9em;
  margin-left: 10px;
`;

export const StopMarker = styled.div`
  float: left;
  margin-left: -36px;
  color: #fff;
`;

export const StopName = styled.div`
  color: #676767;
  margin-top: 3px;
`;

export const StopRow = styled.li`
  z-index: 30;
  position: relative;
`;

export const TransitAlert = styled.a`
  background-color: #eee;
  border-radius: 4px;
  color: #000;
  display: block;
  margin-top: 5px;
  padding: 8px;
  text-decoration: none;
`;

export const TransitAlertBody = styled.div`
  font-size: 12px;
  margin-left: 30px;
  white-space: pre-wrap;
`;

export const TransitAlertEffectiveDate = styled.div`
  margin-top: 5px;
  margin-left: 30px;
  font-size: 12px;
  font-style: italic;
`;

export const TransitAlertHeader = styled.div`
  font-size: 14px;
  margin-left: 30px;
  font-weight: 600;
`;

export const TransitAlertIconContainer = styled.div`
  float: left;
  font-size: 18px;
`;

export const TransitAlerts = styled.div`
  display: block;
  margin-top: 3px;
`;

export const TransitAlertToggle = styled(TransparentButton)`
  color: #d14727;
  cursor: pointer;
  display: inline-block;
  font-weight: 400;
  margin-top: 8px;
  padding: 0;
`;

export const TransitLegDetails = styled.div`
  margin-top: 5px;
`;

export const TransitLegDetailsHeader = styled.div`
  color: #676767;
  display: flex;
`;

export const TransitLegExpandedBody = styled.div`
  font-size: 14px;
`;

export const TransitLegFare = styled.div`
  /* no styling */
`;

export const TransitLegSummary = styled(TransparentButton)`
  padding: 0;
`;

export const DefaultAlertToggleIcon = styled(ExclamationTriangle).attrs({
  size: 15
})``;

export const DefaultAlertBodyIcon = styled(ExclamationTriangle).attrs({
  size: 18
})``;

export const AgencyInfo = styled.div`
  margin-top: 5px;

  a {
    color: #337ab7;
    text-decoration: none;
  }

  a:hover {
    text-decoration: underline;
  }

  img {
    margin-left: 5px;
    vertical-align: middle;
  }
`;
