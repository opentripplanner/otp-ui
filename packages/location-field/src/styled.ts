import styled, { css, keyframes } from "styled-components";
import { Spinner as FASpinner } from "@styled-icons/fa-solid/Spinner";

const hiddenCss = css`
  clip: rect(0, 0, 0, 0);
  display: inline-block;
  height: 0;
  overflow: hidden;
  width: 0;
`;

export const HiddenContent = styled.span`
  ${hiddenCss}
`;

export const BaseButton = styled.button`
  border: none;
  background: none;
`;

export const ClearButton = styled(BaseButton)`
  color: #888;
  cursor: pointer;
  width: 30px;
`;

export const ClearBoth = styled.div`
  clear: both;
`;

export const DropdownButton = styled(BaseButton)`
  width: 30px;
`;

export const MenuItemList = styled.ul.attrs({
  role: "listbox"
})`
  background-clip: padding-box;
  background-color: #fff;
  border-radius: 4px;
  border: 1px solid rgba(0, 0, 0, 0.15);
  box-shadow: 0 6px 12px rgba(0, 0, 0, 0.175);
  float: left;
  font-size: 14px;
  left: 0;
  list-style: none;
  margin: 2px 0 0;
  min-width: 160px;
  padding: 5px 0;
  position: absolute;
  text-align: left;
  top: 100%;
  /* this is an annoyingly high number, but is needed to be on top of some otp-rr components */
  z-index: 1000000;

  /* If the associated input is not in an expanded state, hide the list. */
  input[aria-expanded="false"] ~ & {
    ${hiddenCss}
  }
`;

export const Input = styled.input`
  border: none;
  box-shadow: none;
  font-size: 17px;
  outline: none;
`;

export const InputGroup = styled.div`
  border-bottom: 1px solid #000;
  border-collapse: separate;
  display: table;
  margin-bottom: 15px;
  position: relative;
`;

export const MenuItemA = styled.a<{ active?: boolean }>`
  background-color: ${props => (props.active ? "#337ab7" : "transparent")};
  clear: both;
  color: ${props => (props.active ? "#fff" : "#333")};
  display: block;
  font-weight: 400;
  line-height: 1.42857143;
  padding: 3px 20px;
  text-decoration: none;
  white-space: nowrap;
`;

export const MenuGroupHeader = styled.h2<{
  bgColor?: string;
  fgColor?: string;
}>`
  color: ${props => props.fgColor || "#eee"};
  background-color: ${props => props.bgColor || "#333"};
  font-size: 12px;
  font-weight: normal;
  line-height: 1.42857143;
  margin: 0;
  padding: 0px 10px;
  text-align: center;
  white-space: nowrap;
`;

export const MenuItemLi = styled.li`
  &:hover {
    /* TODO: adjust highlight color based on props.color? */
    background-color: #f5f5f5;
    cursor: pointer;
  }

  /* For disabled (aria-hidden) elements,
     show an arrow pointer (no I-beam) and don't highlight the background. */
  &[aria-hidden="true"]:hover {
    background-color: unset;
    cursor: default;
  }
`;

export const OptionContainer = styled.span`
  display: block;
  padding-top: 5px;
  padding-bottom: 3px;
`;

export const OptionSubTitle = styled.span`
  color: #686868;
  font-size: 12px;
  margin-left: 6px;
`;

export const OptionContent = styled.span`
  margin-left: 30px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
`;

export const OptionIconContainer = styled.span`
  float: left;
`;

export const RouteName = styled.span`
  background-color: gray;
  color: white;
  padding: 2px 3px 0px;
  margin-right: 5px;
`;

export const StaticMenuItemList = styled(MenuItemList)`
  border: none;
  box-shadow: none;
  display: block;
`;

export const StopContentContainer = styled.span`
  margin-left: 30px;
`;

export const StopDistance = styled.span`
  font-size: 8px;
`;

export const StopIconAndDistanceContainer = styled.span`
  float: left;
  padding-top: 3px;
`;

export const StopName = styled.span``;

export const StopRoutes = styled.span`
  font-size: 9px;
`;

const rotateAnimation = keyframes`
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
`;

export const Spinner = styled(FASpinner)`
  animation: ${rotateAnimation} 1.2s linear infinite;
`;
