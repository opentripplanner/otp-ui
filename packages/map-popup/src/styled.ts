import styled from "styled-components";

/* eslint-disable-next-line import/prefer-default-export */
export const ViewStopButton = styled.button<{ stopId?: string }>`
  background: none;
  border-bottom: none;
  ${props =>
    props.stopId
      ? "border-left: 1px solid #000; margin-left: 5px;"
      : "border-left: none; padding-left: 0;"};
  border-right: none;
  border-top: none;
  color: #008;
  font-family: inherit;
  padding-top: 0;
`;
