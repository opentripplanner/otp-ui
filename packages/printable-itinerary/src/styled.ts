import {
  Defaults,
  Styled as ItinBodyClasses
} from "@opentripplanner/itinerary-body";
import styled from "styled-components";

export const Leg = styled.div`
  margin-bottom: 10px;
  border-top: 1px solid grey;
  padding-top: 18px;
  display: flex;
`;

export const CollapsedTop = styled(Leg)`
  border-top: none;
  padding-top: 0;
`;

export const LegBody = styled.div`
  margin-left: 10px;
`;

export const LegDetail = styled.div`
  font-size: 14px;
  margin-top: 3px;
`;

export const LegDetails = styled.div`
  margin-top: 5px;
`;

// TODO Refactor
export const LegHeader = styled.div`
  font-size: 18px;
`;

export const AccessLegDescription = styled(Defaults.AccessLegDescription)`
  font-size: 18px;

  ${ItinBodyClasses.LegDescriptionMode},
  ${ItinBodyClasses.LegDescriptionPlace} {
    font-weight: bold;
  }
`;

export const LegAnnotation = styled.div`
  align-items: center;
  display: flex;
  flex-direction: column;
  height: 100%;
  min-width: 70px;
`;

export const AccessLegStep = styled(Defaults.AccessLegStep)`
  ${ItinBodyClasses.StepStreetName} {
    font-weight: bold;
  }
`;

export const RouteLongName = styled(Defaults.RouteLongName)`
  font-weight: bold;
  ${ItinBodyClasses.LegDescriptionHeadsignPrefix} {
    font-weight: normal;
  }
`;

export const ModeIcon = styled.div`
  float: left;
  width: 32px;
  height: 32px;
`;

export const TripAccessibilityScoreWrapper = styled.div`
  margin-bottom: 1em;
`;

export const PrintableItinerary = styled.div``;
