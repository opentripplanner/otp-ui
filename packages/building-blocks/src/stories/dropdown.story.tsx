import React from "react";
import { ComponentMeta } from "@storybook/react";
import styled from "styled-components";
import Dropdown from "../dropdown";
import blue from "../colors/blue";

const options = [
  { value: "1", label: "One" },
  { value: "2", label: "Two" },
  { value: "3", label: "Three" }
];

const NavItemWrapper = styled.div`
  align-items: center;
  background: ${blue[900]};
  display: flex;
  min-height: 50px;
  padding: 0 10px;
  position: relative;
  width: 100%;

  .navBarItem {
    position: static;
    & > button {
      background: transparent;
      border: none;
      color: white;
      padding: 15px;

      @media (max-width: 768px) {
        padding: 10px;
      }

      &:hover {
        background: rgba(0, 0, 0, 0.05);
        color: #ececec;
      }
    }
  }
`;

export default {
  title: "Building-Blocks/Dropdown",
  component: Dropdown
} as ComponentMeta<typeof Dropdown>;

export const DropdownWithLabel = (): React.ReactElement => (
  <Dropdown
    id="dropdown-with-label"
    label="Dropdown with label"
    listLabel="Dropdown menu"
    text="Dropdown with label"
  >
    More content here
  </Dropdown>
);

export const DropdownWithList = (): React.ReactElement => (
  <Dropdown
    id="dropdown-with-ul"
    label="Dropdown with ul"
    listLabel="Dropdown menu"
    text="Dropdown with ul"
  >
    {options.map(option => (
      <li key={option.value} value={option.value}>
        <button type="button">{option.label}</button>
      </li>
    ))}
  </Dropdown>
);

export const DropdownWithListAlignMenuLeft = (): React.ReactElement => (
  <Dropdown
    alignMenuLeft
    id="dropdown-with-ul"
    label="Dropdown with ul"
    listLabel="Dropdown menu"
    text="Dropdown with ul"
  >
    {options.map(option => (
      <li key={option.value} value={option.value}>
        <button type="button">{option.label}</button>
      </li>
    ))}
  </Dropdown>
);

export const DropdownAsOtprrNavItem = (): React.ReactElement => (
  <NavItemWrapper>
    <Dropdown
      className="navBarItem"
      id="dropdown-navbar"
      label="Dropdown navbar"
      listLabel="Dropdown within navbar"
      text="Icon"
    >
      {options.map(option => (
        <button key={option.value} type="button">
          {option.label}
        </button>
      ))}
    </Dropdown>
  </NavItemWrapper>
);
