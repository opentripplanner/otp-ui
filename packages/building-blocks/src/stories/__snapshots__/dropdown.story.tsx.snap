// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Building-Blocks/Dropdown DropdownAsOtprrNavItem smoke-test 1`] = `
<div class="dropdownstory__NavItemWrapper-sc-giygrt-0 fXLCFd">
  <span class="styled__DropdownWrapper-sc-kes9ha-2 dgncpu navBarItem"
        id="dropdown-navbar-wrapper"
  >
    <button aria-expanded="false"
            aria-haspopup="listbox"
            aria-label="Dropdown navbar"
            id="dropdown-navbar-label"
            title="Dropdown navbar"
            class="styled__DropdownButton-sc-kes9ha-0 eBZyNH"
    >
      <span>
        Icon
      </span>
      <span class="caret"
            role="presentation"
      >
      </span>
    </button>
  </span>
</div>
`;

exports[`Building-Blocks/Dropdown DropdownWithLabel smoke-test 1`] = `
<span class="styled__DropdownWrapper-sc-kes9ha-2 dgncpu"
      id="dropdown-with-label-wrapper"
>
  <button aria-expanded="false"
          aria-haspopup="listbox"
          aria-label="Dropdown with label"
          id="dropdown-with-label-label"
          title="Dropdown with label"
          class="styled__DropdownButton-sc-kes9ha-0 eBZyNH"
  >
    <span>
      Dropdown with label
    </span>
    <span class="caret"
          role="presentation"
    >
    </span>
  </button>
</span>
`;

exports[`Building-Blocks/Dropdown DropdownWithList smoke-test 1`] = `
<span class="styled__DropdownWrapper-sc-kes9ha-2 dgncpu"
      id="dropdown-with-ul-wrapper"
>
  <button aria-expanded="false"
          aria-haspopup="listbox"
          aria-label="Dropdown with ul"
          id="dropdown-with-ul-label"
          title="Dropdown with ul"
          class="styled__DropdownButton-sc-kes9ha-0 eBZyNH"
  >
    <span>
      Dropdown with ul
    </span>
    <span class="caret"
          role="presentation"
    >
    </span>
  </button>
</span>
`;

exports[`Building-Blocks/Dropdown DropdownWithListAlignMenuLeft smoke-test 1`] = `
<span class="styled__DropdownWrapper-sc-kes9ha-2 dgncpu"
      id="dropdown-with-ul-wrapper"
>
  <button aria-expanded="false"
          aria-haspopup="listbox"
          aria-label="Dropdown with ul"
          id="dropdown-with-ul-label"
          title="Dropdown with ul"
          class="styled__DropdownButton-sc-kes9ha-0 eBZyNH"
  >
    <span>
      Dropdown with ul
    </span>
    <span class="caret"
          role="presentation"
    >
    </span>
  </button>
</span>
`;
